src/conn.rs

```diff
@@ -787,7 +794,16 @@ impl<const N: usize, P: ConnectionPeer> Connection<N, P> {
         // NOTE: We should probably move this so that multiple incoming packets can be processed
         // before we send a STATE.
         match packet.packet_type() {
-            PacketType::Syn | PacketType::Fin | PacketType::Data => {
+            PacketType::Syn => {
+                if let Some(state) = &self.syn_ack {
+                    let event = SocketEvent::Outgoing((state.clone(), self.peer.clone()));
```

Comment:
```suggestion
                if let Some(state) = self.syn_ack.take() {
                    let event = SocketEvent::Outgoing((state, self.peer.clone()));
```

If we only use this option once, couldn't we just use take? to avoid the clone?. You probably have more context if this would make sense or not, but it looking reasonable to use take instead

Reply:
I would say that we don't want to `take()`. Maybe this response is also not received, and we have to do it again, or something like that.

Also, if I'm not mistaken, this packet should be small, so it shouldn't be too costly to clone.